core/dbt/task/run.py

```diff
@@ -300,26 +301,53 @@ def _build_run_model_result(self, model, context):
             message=str(result.response),
             adapter_response=adapter_response,
             failures=result.get("failures"),
+            batch_results=None,
         )
 
     def _build_run_microbatch_model_result(
         self, model: ModelNode, batch_run_results: List[RunResult]
     ) -> RunResult:
-        failures = sum([result.failures for result in batch_run_results if result.failures])
+        batch_results = BatchResults()
+        for result in batch_run_results:
+            if result.batch_results is not None:
+                batch_results += result.batch_results
+            else:
+                # TODO: Should we raise an error here?
```

Comment:
when would we encounter this scenario?

Reply:
We shouldn't, but unfortunately the problem is we can (technically). The function takes in a list of `RunResult` objects. Additionally, `batch_results` is _optional_ on a `RunResult`. So we don't "technically" have any guarantee that `batch_results` will always be a `BatchResults` object. Write now, how the `RunResults` are constructed for microbatch batches, the `batch_results` should _never_ be `None`. If we had a subclass of `RunResult` just for microbatch settings, we could guarantee the `batch_results`. But as we haven't, we thus live with this technical possibility.